src/client/pydaos/pydaos_core.py

```diff
@@ -154,6 +159,30 @@ def rootkv(self, cid=ObjClassID.OC_SX):
         oid = ObjID(hi, lo)
         return KVObj(self.coh, oid, self)
 
+    def get_kv_by_name(self, name, root=None, create=False):
+        """Return KV by name.
+
+        Allow selection of root (or parent) container, and
+        optionally create kv if not found"""
+
+        if not root:
+            root = self.rootkv()
+        if name in root:
+            object_data = pickle.loads(root[name])
+            return self.kv(object_data['oid'])
+
+        if not create:
+            raise KvNotFound
```

Comment:
Should this error indicate which `name` was not found when not creating?

Reply:
Ok, I added this which works, and I added a test for it.

It is causing some problems however, it seems that when I do that there's an extra reference taken on the container somehow so the node_local_test code doesn't try to close the container until after it's stopped the DAOS server.  This wouldn't be an issue in normal use as the daos server wouldn't be a child of the python process but it does make it difficult to add a test case for it using the current technique.